### PR TITLE
fix: Include git commit id while building docker image if possible

### PR DESCRIPTION
**Pull request by KShivendu**
Try to include git commit ID while building docker image if .git is present. Otherwise, show a warning.

### All Submissions:

* [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### CODE REVIEW

**Review by timvisee — COMMENTED** [review]
I think this is obsolete, since the ref is already exposed here (the second one) in each job:

https://github.com/qdrant/qdrant/actions/runs/7566398392/job/20603708488?pr=3419#step:3:77

> HEAD is now at 6bb2a8c Merge b0f71218a9e9e66a01495d90c990cdb128180969 into fed3262048eaef6ed4a4ab520c00cc459dbf2dcb

File: lib/api/build.rs [hunk 1]
```diff
@@ -27,8 +27,12 @@ fn main() -> std::io::Result<()> {
...
     match Command::new("git").args(["rev-parse", "HEAD"]).output() {
         Ok(output) => {
-            let git_commit_id = str::from_utf8(&output.stdout).unwrap().trim();
-            println!("cargo:rustc-env=GIT_COMMIT_ID={git_commit_id}");
+            if output.status.success() {
+                let git_commit_id = str::from_utf8(&output.stdout).unwrap().trim();
+                println!("cargo:rustc-env=GIT_COMMIT_ID={git_commit_id}");
+            } else {
+                println!("cargo:warning=current git commit hash could not be determined");
+            }
```
Comment (KShivendu) [inline lib/api/build.rs:35]: @timvisee the warning didn't work previously because we need to check success status of the subprocess as well.

**Review by timvisee — APPROVED** [review]

**Comment by timvisee** [comment]
Feel free to hit the green button once CI is green! :green_circle:

**Comment by generall** [comment]
This PR breaks CI on arm container build, I am reverting it.

**Pull request merged**